src/token-service/metadata-token-service.ts

```diff
@@ -67,12 +67,12 @@ export class MetadataTokenService implements TokenService {
                 `failed to fetch token from metadata service: ${lastError}`,
             );
         }
-        setTimeout(async () => {
+        setInterval(async () => {
```

Comment:
You can not spawn interval handlers and not clearing them when the SDK will be destroyed.